lib/grammers-client/src/client/net.rs

```diff
@@ -278,6 +299,84 @@ impl Client {
         }
     }
 
+    async fn export_authorization(
+        &self,
+        target_dc_id: i32,
+    ) -> Result<tl::types::auth::ExportedAuthorization, InvocationError> {
+        let request = tl::functions::auth::ExportAuthorization {
+            dc_id: target_dc_id,
+        };
+        return match self.invoke(&request).await {
+            Ok(tl::enums::auth::ExportedAuthorization::Authorization(exported_auth)) => {
+                Ok(exported_auth)
+            }
+            Err(e) => Err(e),
+        };
+    }
+
+    async fn connect_sender(&self, dc_id: i32) -> Result<Arc<FileDownloader>, InvocationError> {
+        let mut mutex = self.0.downloader_map.write().await;
+        debug!("Connecting new datacenter {}", dc_id);
+        return match connect_sender(dc_id, &self.0.config).await {
+            Ok((new_sender, new_tx)) => {
+                let new_downloader = Arc::new(FileDownloader {
+                    sender: AsyncMutex::new("dc_sender", new_sender),
+                    request_tx: Mutex::new("dc_enqueuer", new_tx),
+                    stepping_done: Notify::new(),
+                });
+
+                // export auth
+                let authorization = self.export_authorization(dc_id).await?;
+
+                // import into new sender
+                let request = tl::functions::auth::ImportAuthorization {
+                    id: authorization.id,
+                    bytes: authorization.bytes,
+                };
+                self.invoke_request(
+                    &request,
+                    &new_downloader.request_tx,
+                    &new_downloader.sender,
+                    &new_downloader.stepping_done,
+                )
+                .await?;
+
+                mutex.insert(dc_id, new_downloader.clone());
+                Ok(new_downloader.clone())
+            }
+            Err(e) => panic!("Cannot obtain new sender to datacenter {}", e),
+        };
+    }
+
+    async fn get_downloader(
+        &self,
+        dc_id: i32,
+    ) -> Result<Option<Arc<FileDownloader>>, InvocationError> {
+        return Ok({
+            let guard = self.0.downloader_map.read().await;
+            guard.get(&dc_id).map(|f| f.clone())
+        });
+    }
+
+    pub async fn invoke_in_dc<R: tl::RemoteCall>(
```

Comment:
I think it's fine to leave this as public (there's other requests which must be executed in a different datacenter, like getting graph data from channels I think).